### PR TITLE
SPCTR-3369 [Extension] Quick Action Sidebar - POC

### PR DESCRIPTION
**Pull request by adi3890**
### Description
- Adding a feature of Quick Action Sidebar in editor
which helps user to quickly use core spectra blocks ( will be extended further ) while creating a post 

### Checklist:
- [ ] My code is tested
- [ ] My code follows accessibility standards 
- [ ] My code has proper inline documentation 
- [ ] I've included any necessary tests 
- [ ] I've included developer documentation 
- [ ] I've added proper labels to this pull request

### CODE REVIEW

File: src/blocks/extensions/quick-action-sidebar/editor.lazy.scss [hunk 1]
```diff
@@ -0,0 +1,89 @@
+// All Zip AI Sidebar Editor CSS.
```
Comment (AbhijitNage123) [inline src/blocks/extensions/quick-action-sidebar/editor.lazy.scss:1]: @adi3890 Comment should be updated to quick action sidebar

**Review by AbhijitNage123 — APPROVED** [review]
PR look good to me. Some minor comment string changes. Also make sure we added changelog entry for same

File: src/blocks/extensions/quick-action-sidebar/editor.lazy.scss (hunk 1 above)
Comment (adi3890) [reply src/blocks/extensions/quick-action-sidebar/editor.lazy.scss:1]: done

File: src/blocks/extensions/quick-action-sidebar/editor.lazy.scss [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+				&--icon {
+					display: flex;
+					width: 24px;
+					height: 24px;
+					justify-content: center;
+					align-items: center;
+
+					&--name {
+						.components-popover__content {
```
Comment (bsf-zanev) [inline src/blocks/extensions/quick-action-sidebar/editor.lazy.scss:53]: `.spectra-ee-quick-access__sidebar--blocks--block--icon--name` is not as per the [BEM Convention](https://getbem.com/naming/) - and will be confusing to maintain.

File: src/blocks/extensions/quick-action-sidebar/index.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
+/**
+ * The Quick Access React App for Spectra Enhanced Editor.
+ */
+import domReady from '@wordpress/dom-ready';
+import { createRoot } from 'react-dom';
+import Sidebar from './components/sidebar.js';
+const { enableQuickActionSidebar } = uagb_blocks_info;
+
+domReady( function () {
```
Comment (bsf-zanev) [inline src/blocks/extensions/quick-action-sidebar/index.js:9]: Minor suggestion.

```suggestion
domReady( () => {
```

File: src/blocks/extensions/quick-action-sidebar/index.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+	}
+	const interval = setInterval( () => {
+		const rootElement = document.querySelector( '.interface-interface-skeleton__body' );
+		if ( ! rootElement ) {
+			return;
+		}
+
+		clearInterval( interval );
+
+		let container =document.querySelector( '.interface-interface-skeleton__body .spectra-ee-quick-access' );
```
Comment (bsf-zanev) [inline src/blocks/extensions/quick-action-sidebar/index.js:21]:
```suggestion
		let container = rootElement.querySelector( '.spectra-ee-quick-access' );
```

File: src/blocks/extensions/quick-action-sidebar/index.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+		if ( ! container ) {
+			const parentDiv = rootElement.insertBefore( document.createElement( 'div' ), rootElement.firstChild );
+			container = parentDiv.appendChild( document.createElement( 'div' ) );
+			container.classList.add( 'spectra-ee-quick-access' );
+		}
+		const root = createRoot( container );
+
+		root.render(
+			<Sidebar/>
+		);
```
Comment (bsf-zanev) [inline src/blocks/extensions/quick-action-sidebar/index.js:32]: Minor Suggestion.

```suggestion
		root.render( <Sidebar/> );
```

File: admin-core/ajax/common-settings.php [hunk 6]
```diff
@@ -437,6 +438,18 @@ public function enable_masonry_gallery() {
 		$this->save_admin_settings( 'uag_enable_masonry_gallery', sanitize_text_field( $value ) );
 	}
 
+	/**
+	 * Save setting - Enables quick action sidebar.
+	 *
+	 * @return void
+	 */
```
[the remark this answers is not in the document]
Comment (adi3890) [reply admin-core/ajax/common-settings.php:445]: done

File: admin-core/assets/src/dashboard-app/pages/blocks/QuickActionSidebarExtension.js [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+import { useSelector, useDispatch } from 'react-redux';
+import { Switch } from '@headlessui/react'
+import UAGB_Block_Icons from '@Common/block-icons';
+import { useEffect } from 'react';
+
+import getApiData from '@Controls/getApiData';
+
+function classNames( ...classes ) {
+    return classes.filter( Boolean ).join( ' ' )
+}
```
[the remark this answers is not in the document]
Comment (adi3890) [reply admin-core/assets/src/dashboard-app/pages/blocks/QuickActionSidebarExtension.js:11]: done

File: classes/class-uagb-init-blocks.php [hunk 8]
```diff
@@ -730,75 +730,101 @@ function( $a, $b ) {
...
+					'uagb/icon',
+					'uagb/buttons',
+					'uagb/info-box',
+					'uagb/call-to-action',
+					'uagb/countdown',
+				)
+			);
+
+			$quick_sidebar_allowed_blocks = \UAGB_Admin_Helper::get_admin_settings_option( 'uagb_quick_sidebar_allowed_blocks', $default_allowed_quick_sidebar_blocks );
+			$quick_sidebar_allowed_blocks = ! empty( $quick_sidebar_allowed_blocks ) && is_array( $quick_sidebar_allowed_blocks ) ? $quick_sidebar_allowed_blocks : $default_allowed_quick_sidebar_blocks;
```
[the remark this answers is not in the document]
Comment (adi3890) [reply classes/class-uagb-init-blocks.php:817]: added a check

File: admin-core/assets/src/dashboard-app/pages/blocks/QuickActionSidebarExtension.js [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+
+const QuickActionSidebarExtension = () => {
+
+    const enableQuickActionSidebarExtension = useSelector( ( state ) => state.enableQuickActionSidebarExtension );
+    const dispatch = useDispatch();
+
+    const QuickActionSidebarStatus = 'disabled' === enableQuickActionSidebarExtension ? false : true;
```
[the remark this answers is not in the document]
Comment (adi3890) [reply admin-core/assets/src/dashboard-app/pages/blocks/QuickActionSidebarExtension.js:18]: this need to be withing brackets, done

File: admin-core/assets/src/dashboard-app/pages/blocks/QuickActionSidebarExtension.js [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+                { UAGB_Block_Icons['masonry-gallery'] }
+            </div>
+            <div className="uagb-admin-block__extension-title flex-1 min-w-0">
+                <p className="text-base font-medium text-slate-800">
+                    { __( 'Quick Action Bar', 'ultimate-addons-for-gutenberg' ) }
+                    <div className="inline-block align-text-bottom max-h-4 px-1.5 py-[3px] ml-1.5 text-[10px] leading-[10px] border border-slate-400 text-slate-500 rounded spectra-admin__block-label">
+                        { __( 'Extension', 'ultimate-addons-for-gutenberg' ) }
+                    </div>
+                </p>
+                {/* <a className="focus-visible:text-slate-500 active:text-slate-500 hover:text-slate-500 focus:text-slate-400 text-slate-400 text-sm truncate" href='https://wpspectra.com/docs/quick-action-bar/' target="_blank"rel="noreferrer">{__( 'Documentation', 'ultimate-addons-for-gutenberg' )}</a> */}
```
[the remark this answers is not in the document]
Comment (adi3890) [reply admin-core/assets/src/dashboard-app/pages/blocks/QuickActionSidebarExtension.js:71]: done

File: src/blocks/extensions/quick-action-sidebar/index.js (hunk 4 above)
Comment (adi3890) [reply src/blocks/extensions/quick-action-sidebar/index.js:21]: done

File: src/blocks/extensions/quick-action-sidebar/index.js (hunk 3 above)
Comment (adi3890) [reply src/blocks/extensions/quick-action-sidebar/index.js:9]: done

File: src/blocks/extensions/quick-action-sidebar/components/draggable-block.js [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+					onDragEnd={ () => {
+					}  }
+					onMouseOver={ handleMouseOver }
+					onMouseOut={ handleMouseOut }
+					onFocus={ handleMouseOver }
+					onBlur={ handleMouseOut }
+				>
+						<div className='spectra-ee-quick-access__sidebar--blocks--block--icon'>
+
+								<Icon icon={ block.icon && block.icon.src ? block.icon.src : block.icon } />
```
[the remark this answers is not in the document]
Comment (adi3890) [reply src/blocks/extensions/quick-action-sidebar/components/draggable-block.js:50]: added

**Review by bsf-zanev — APPROVED** [review]

**Pull request merged**